Implement prefix for saved mapping file

MindaugasLaganeckas opened the pull request:
Address #1037

StefH commented on src/WireMock.Net/Serialization/MappingFileNameSanitizer.cs:
```diff
@@ -0,0 +1,49 @@
+/// </summary>
+public class MappingFileNameSanitizer
+{
+    private readonly char _replaceChar = '_';
```
private const char ReplaceChar = '_';

StefH commented on test/WireMock.Net.Tests/Serialization/MappingFileNameSanitizerTests.cs:
```diff
@@ -0,0 +1,136 @@
+
+public class MappingFileNameSanitizerTests
+{
+    private readonly string _mappingGuid = "ce216a13-e7d6-42d7-91ac-8ae709e2add1";
```
both can be const

StefH commented on test/WireMock.Net.Tests/Serialization/MappingFileNameSanitizerTests.cs:
```diff
@@ -0,0 +1,136 @@
+public class MappingFileNameSanitizerTests
+{
+    private readonly string _mappingGuid = "ce216a13-e7d6-42d7-91ac-8ae709e2add1";
+    private readonly string _mappingTitle =
```
place this on 1 line

MindaugasLaganeckas commented:
Thank you for the feedback! I have addressed all comments and added an extra test for the usecase, that is going to be our primary (somehow missed in the initial implementation).

StefH commented on src/WireMock.Net/Serialization/MappingFileNameSanitizer.cs:
```diff
@@ -0,0 +1,49 @@
+    /// </summary>
+    public string BuildSanitizedFileName(IMapping mapping)
+    {
+        const string mappingTitlePrefix = "Proxy Mapping for ";
```
Use the public const value in the ProxyAndRecordSettings class here.

StefH commented on src/WireMock.Net/Settings/ProxyAndRecordSettings.cs:
```diff
@@ -95,9 +95,14 @@ public string SaveMappingForStatusCodePattern
+    /// <summary>
+    /// Set prefix for saved mapping file.
+    /// </summary>
+    public string PrefixForSavedMappingFile { get; set; } = "Proxy Mapping for ";
```
Create a public const for this the value in this class.

StefH commented on test/WireMock.Net.Tests/Serialization/MappingFileNameSanitizerTests.cs:
```diff
@@ -0,0 +1,160 @@
+        mappingMock.Setup(m => m.Title).Returns(MappingTitle);
+        mappingMock.Setup(m => m.Guid).Returns(new Guid(MappingGuid));
+
+        var settingsMock = new WireMockServerSettings
```
officially, this is not mocked, so the name can be `settings`

MindaugasLaganeckas commented:
@StefH if it is not too much trouble, may I ask you to release the feature? I would like to start using it as soon as possible  😄

StefH commented:
New NuGet will be released today or tomorrow.

StefH commented:
NuGet is released

MindaugasLaganeckas commented:
I will switch to the latest Wiremock.Net right away for my demo today. Thank you for expediting the release! 🎅

StefH commented:
Ah cool. A demo for your company?

MindaugasLaganeckas commented:
More like a demo for the team inside the company 😄
